src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/1.sql

```diff
@@ -227,6 +227,52 @@ CREATE CLUSTERED INDEX IXC_LastModifiedClaim on dbo.ResourceWriteClaim
     ClaimTypeId
 )
 
+/*************************************************************
+    Compartments
+**************************************************************/
+
+CREATE TABLE dbo.CompartmentType
+(
+    CompartmentTypeId tinyint IDENTITY(1,1) NOT NULL,
+    Name varchar(128) NOT NULL
+)
+
+CREATE UNIQUE CLUSTERED INDEX IXC_CompartmentType on dbo.CompartmentType
+(
+    Name
+)
+
+CREATE TYPE dbo.CompartmentAssignmentTableType_1 AS TABLE  
+(
+    CompartmentTypeId tinyint NOT NULL,
+    ReferenceResourceId varchar(64) NOT NULL
+)
+
+CREATE TABLE dbo.CompartmentAssignment
+(
+    ResourceSurrogateId bigint NOT NULL,
+    CompartmentTypeId tinyint NOT NULL,
+    ReferenceResourceId varchar(64) NOT NULL,
+    IsHistory bit NOT NULL,
+) WITH (DATA_COMPRESSION = PAGE)
+
+CREATE CLUSTERED INDEX IXC_CompartmentAssignment 
```

Comment:
Should this be unique?

Reply:
In general, we are not doing anything at the moment to ensure that duplicates are not present in the resource and/or what the extractors yield. For example, for an observation resource, the same practitioner could appear twice in the list of performers. Or a patient could be named "John John", in which case the extracted name parameter would have "John" twice. 

Having duplicates does not bring any benefit and forces to do a DISTINCT clause when searching. 
So we probably should make sure everything is unique and filter out duplicates before adding them, but when it gets to strings, it has to be unique respecting the same collation rules as the DB column. I am deferring this to later, once we have sorted out the collation configuration. Sound ok?

Reply:
Yes, sounds okay.